### ISSUE TITLE
This is how browser UI & user-features are done to help a user navigate hostile web environment

### ISSUE THREAD

**Issue by v-fox** [mozilla/tofino#146]
![jamin_on_dual-screen](https://cloud.githubusercontent.com/assets/362763/14586876/903a01d4-04c8-11e6-99b7-92102d5f74da.jpg)
![screenshot_1](https://cloud.githubusercontent.com/assets/362763/14586877/903e7534-04c8-11e6-9419-cc47025bd393.png)

This is Firefox from [my livecd project](https://github.com/v-fox/live_opensuse_hsf).
It is similar to what I personally use and [moded with addons](https://github.com/v-fox/live_opensuse_hsf/tree/master/source/root/home/hacker/.mozilla/firefox/y30d7fz5.hacker) for:
* tree-style tabbar which makes dealing with hundreds of tabs easy;
* button for easy switching between direct connection and localhost-proxy (privoxy) with Tor/I2P redirection (would be cool to have that per-tab), helps nicely with MitM-based state censorship;
* DNS redirection while proxy is active with `network.proxy.socks_remote_dns` option & system-wide usage of unbound/[dnscrypt-proxy](https://github.com/jedisct1/dnscrypt-proxy) for round-robin DNS requests via tunnels to trusted DNS servers (outside of the country);
* referrer & canvas & geolocation fakers for non-whitelisted sites and script & advertisement & "social" blockers to increase performance by getting rid of garbage and opt-out from being a walking social experiment for Big Data companies (I would rather trust anti-constitutionally censorship-loving government with my "social and demographic status" than private international corporations, thank you very much);
* forced element-blocking and right-click de-deactivation;
* networking & geolocation data about server of a site and the local machine;
* more explicit loading status;
* colourful icons, more icons in general but with background-darkening/font-whitening for ease of visual distinction with decreased eye strain;
* support for Chrome-flash with per-element blocking for things that require it;
* attempt in making downloads less sucky (doesn't save from Firefox's dumb insistence on redownloading files in their entirety);
* media downloading for usage of proper media player, like [mpv](https://github.com/mpv-player/mpv) with its advanced scaling & interpolation algorithms, colour correction, debanding/dithering and such;
* better session management & saving;
* easier bookmarking;
* easier password database management & access;
* easier cookie destruction;
* more options for page saving and screenshot-making;
* page-translation and Google Noto fonts for all languages.

On the right side of the first picture you can see JAMin, JACK "mastering" tool from the second monitor which works greatly as global sound compressor & equalizer which requires PulseAudio->JACK redirection or native support for JACK. I patch FF for newer version of libcubeb to force it to use JACK natively but also use PA configs that make JACK default PA sink & source and not let it to get ahold of "primary" sound device that is reserved for JACK.

In short, I used FF my whole life and it was not easy to make it bearable but for anything else it's completely impossible nowadays. Please, don't make it into another UI-castrated user-spying Chrome & IE knock-off. Moreover, Web isn't something users should or could trust. If you can't or don't want to do it right, leave it for addon-makers.

**Comment by victorporof** [comment]
Will consider your suggestion. @phlsa please take a look at this.
In the meantime I will close this issue until we have a proper medium for UI/UX suggestions.
